src/Controls/src/Core/LegacyLayouts/Layout.cs

```diff
@@ -341,10 +341,12 @@ protected virtual void InvalidateLayout()
 		[Obsolete("Use ArrangeOverride")]
 		protected abstract void LayoutChildren(double x, double y, double width, double height);
 
-		internal override void OnChildMeasureInvalidatedInternal(VisualElement child, InvalidationTrigger trigger)
+		internal override void OnChildMeasureInvalidatedInternal(VisualElement child, InvalidationTrigger trigger, int depth)
 		{
+			CurrentInvalidationDepth = depth;
 			// TODO: once we remove old Xamarin public signatures we can invoke `OnChildMeasureInvalidated(VisualElement, InvalidationTrigger)` directly
 			OnChildMeasureInvalidated(child, new InvalidationEventArgs(trigger));
```

Comment:
can't we pass the depth as part of the InvalidationEventArgs ? I'm always afraid of keeping state...

Reply:
Yea, that works way better!

I wasn't happy with keeping state either :-/ 

updated PR